tests/components/seventeentrack/test_sensor.py

```diff
@@ -288,169 +169,150 @@ async def test_friendly_name_changed(hass: HomeAssistant) -> None:
 
         persistent_notification_mock.create.assert_called()
         assert not hass.states.async_entity_ids()
 
 
-async def test_summary_correctly_updated(hass: HomeAssistant) -> None:
+async def test_summary_correctly_updated(
+    hass: HomeAssistant, freezer: FrozenDateTimeFactory, mock_seventeentrack: AsyncMock
+) -> None:
     """Ensure summary entities are not duplicated."""
-    package = Package(
-        tracking_number="456",
-        destination_country=206,
-        friendly_name="friendly name 1",
-        info_text="info text 1",
-        location="location 1",
-        timestamp="2020-08-10 10:32",
-        origin_country=206,
-        package_type=2,
-        status=30,
-    )
-    ProfileMock.package_list = [package]
+    package = get_package(status=30)
+    mock_seventeentrack.return_value.profile.packages.return_value = [package]
+    mock_seventeentrack.return_value.profile.summary.return_value = DEFAULT_SUMMARY
 
-    await _setup_seventeentrack(hass, summary_data=DEFAULT_SUMMARY)
+    await init_integration(hass, VALID_CONFIG_FULL)
 
     assert len(hass.states.async_entity_ids()) == 8
     for state in hass.states.async_all():
         if state.entity_id == "sensor.seventeentrack_package_456":
             break
         assert state.state == "0"
 
-    assert (
-        len(
-            hass.states.get(
-                "sensor.seventeentrack_packages_ready_to_be_picked_up"
-            ).attributes["packages"]
-        )
-        == 1
+    state_ready_picked = hass.states.get(
+        "sensor.seventeentrack_packages_ready_to_be_picked_up"
     )
+    assert state_ready_picked is not None
+    assert len(state_ready_picked.attributes["packages"]) == 1
 
-    ProfileMock.package_list = []
-    ProfileMock.summary_data = NEW_SUMMARY_DATA
+    mock_seventeentrack.return_value.profile.packages.return_value = []
+    mock_seventeentrack.return_value.profile.summary.return_value = NEW_SUMMARY_DATA
 
-    await _goto_future(hass)
+    await _goto_future(hass, freezer)
 
     assert len(hass.states.async_entity_ids()) == 7
     for state in hass.states.async_all():
         assert state.state == "1"
 
+    state_ready_picked = hass.states.get(
+        "sensor.seventeentrack_packages_ready_to_be_picked_up"
+    )
+    assert state_ready_picked is not None
+    assert state_ready_picked.attributes["packages"] is None
+
+
+async def test_summary_error(
+    hass: HomeAssistant, mock_seventeentrack: AsyncMock
+) -> None:
+    """Test summary empty if error."""
+    package = get_package(status=30)
+    mock_seventeentrack.return_value.profile.packages.return_value = [package]
+    mock_seventeentrack.return_value.profile.summary.side_effect = SeventeenTrackError(
+        "Error"
+    )
+
+    await init_integration(hass, VALID_CONFIG_FULL)
+
+    assert len(hass.states.async_entity_ids()) == 1
+    for state in hass.states.async_all():
+        if state.entity_id == "sensor.seventeentrack_package_456":
```

Comment:
Please don't use if statements in tests

Reply:
I see that we already had that before, can you maybe take a stab at removing them?

Reply:
removed